clm/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM nvcr.io/nvidia/nemo:24.07
+WORKDIR /app
+
+# GCSfuse components (used to provide shared storage, not intended for high performance)
+RUN apt-get update && apt-get install --yes --no-install-recommends \
+    ca-certificates \
+    curl \
+    gnupg \
+  && echo "deb https://packages.cloud.google.com/apt gcsfuse-buster main" \
+    | tee /etc/apt/sources.list.d/gcsfuse.list \
+  && echo "deb https://packages.cloud.google.com/apt cloud-sdk main" \
+    | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list \
+  && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - \
+  && apt-get update \
+  && apt-get install --yes gcsfuse \
+  && apt-get install --yes google-cloud-cli \
+  && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+  && mkdir /gcs
+
+RUN pip install git+https://github.com/NVIDIA/dllogger#egg=dllogger
+RUN pip install datasets==2.20.0
+
+# Install the Google Cloud SDK
+RUN apt-get update && apt-get install -y google-cloud-sdk vim
+
+RUN pip install hydra-core
+RUN pip install sentencepiece
+RUN pip install "git+https://github.com/mlperf/logging.git"
+
+# checkpoint loading in gcs
+RUN pip install gcsfs
+
+COPY *.py /app/
```

Comment:
Propose to change to 

```
# clone current repo
ARG GIT_REF=main
RUN git clone https://github.com/ZhiyuLi-goog/MoE_study.git && \
    cd MoE_study && \
    echo GIT_REF=${GIT_REF} && \
    git checkout ${GIT_REF}
```

So that we can change branch in the docker image for further convenience.